public/app/core/components/sidemenu/SignIn.tsx

```diff
@@ -4,15 +4,21 @@ import { StoreState } from 'app/types';
 import { Icon } from '@grafana/ui';
 
 export const SignIn: FC<any> = ({ url }) => {
-  const loginUrl = `login?redirect=${encodeURIComponent(url)}`;
+  const splitted = url.split('?');
+  let params = new URLSearchParams('');
+  if (splitted.length > 1) {
+    params = new URLSearchParams(splitted[1]);
+  }
+  params.set('forceLogin', 'true');
+  const forcedLoginUrl = splitted[0] + '?' + params.toString();
```

Comment:
nit: Maybe we could extract this code to an external function called `getForcedLoginUrl`, WDYT?

nit: Also, we could look into using the URL (https://developer.mozilla.org/en-US/docs/Web/API/URL) object instead of doing all this split and concatination ourselves. But the tricky part there is what to use as base url  (2:nd param) as url passed to this component might be relative?